runtime/lua/vim/treesitter/_fold.lua

```diff
@@ -2,109 +2,108 @@ local api = vim.api
 
 local M = {}
 
---- Memoizes a function based on the buffer tick of the provided bufnr.
---- The cache entry is cleared when the buffer is detached to avoid memory leaks.
----@generic F: function
----@param fn F fn to memoize, taking the bufnr as first argument
----@return F
-local function memoize_by_changedtick(fn)
-  ---@type table<integer,{result:any,last_tick:integer}>
-  local cache = {}
-
-  ---@param bufnr integer
-  return function(bufnr, ...)
-    local tick = api.nvim_buf_get_changedtick(bufnr)
-
-    if cache[bufnr] then
-      if cache[bufnr].last_tick == tick then
-        return cache[bufnr].result
-      end
-    else
-      local function detach_handler()
-        cache[bufnr] = nil
-      end
-
-      -- Clean up logic only!
-      api.nvim_buf_attach(bufnr, false, {
-        on_detach = detach_handler,
-        on_reload = detach_handler,
-      })
-    end
-
-    cache[bufnr] = {
-      result = fn(bufnr, ...),
-      last_tick = tick,
-    }
+local function foldupdate()
+  -- 'zx' does too much and has buggy behaviour with the cursor.
+  -- Instead simply assign foldmethod which internally calls foldUpdateAll()
```

Comment:
Sounds like we could use a `nvim_*_fold` API... (Not in this PR, of course.)

Reply:
(or use FFI 😬 )

Reply:
tbh, since this is core, we could just add some cheap unstable bindings.

Reply:
Actually, we need more low effort bindings to use in `runtime/`. Let's go that route.